blocktx/peer_settings.go

```diff
@@ -0,0 +1,54 @@
+package blocktx
+
+import (
+	"fmt"
+	"net/url"
+
+	"github.com/spf13/viper"
+)
+
+type Peer struct {
+	Host string
+	Port PeerPort `mapstructure:"port"`
+}
+
+type PeerPort struct {
+	P2P int `mapstructure:"p2p"`
+	ZMQ int `mapstructure:"zmq"`
+}
+
+func (p Peer) GetZMQUrl() (*url.URL, error) {
+	if p.Port.ZMQ == 0 {
+		return nil, fmt.Errorf("port_zmq not set for peer %s", p.Host)
+	}
+
+	if p.Host == "" {
+		return nil, fmt.Errorf("host not set for peer %s", p.Host)
+	}
+
+	zmqURLString := fmt.Sprintf("zmq://%s:%d", p.Host, p.Port.ZMQ)
+
+	return url.Parse(zmqURLString)
+}
+
+func (p Peer) GetP2PUrl() (string, error) {
+	if p.Port.P2P == 0 {
+		return "", fmt.Errorf("port_p2p not set for peer %s", p.Host)
+	}
+
+	if p.Host == "" {
+		return "", fmt.Errorf("host not set for peer %s", p.Host)
+	}
+
+	return fmt.Sprintf("%s:%d", p.Host, p.Port.P2P), nil
+}
+
+func GetPeerSettings() ([]Peer, error) {
+	var peers []Peer
+	err := viper.UnmarshalKey("peers", &peers)
```

Comment:
In my opinion is cleaner to validate/format the settings when they are loaded in `GetPeerSettings()`. This way `GetZMQUrl()` and `GetP2PUrl()` are not necessary.